internal/commands/vaultsecrets/integrations/create.go

```diff
@@ -0,0 +1,248 @@
+		IO:      ctx.IO,
+		Output:  ctx.Output,
+
+		PreviewClient: preview_secret_service.New(ctx.HCP, nil),
+		Client:        secret_service.New(ctx.HCP, nil),
+	}
+
+	cmd := &cmd.Command{
+		Name:      "create",
+		ShortHelp: "Create a new integration.",
+		LongHelp: heredoc.New(ctx.IO).Must(`
+		The {{ template "mdCodeOrBold" "hcp vault-secrets integrations create" }} command creates a new Vault Secrets integration.
+		`),
+		Examples: []cmd.Example{
+			{
+				Preamble: `Create a new Vault Secrets integration:`,
+				Command: heredoc.New(ctx.IO, heredoc.WithPreserveNewlines()).Must(`
+				$ hcp vault-secrets integrations create sample-integration --config-file=path-to-file/config.yaml
+				`),
+			},
+		},
+		Args: cmd.PositionalArguments{
+			Args: []cmd.PositionalArgument{
+				{
+					Name:          "NAME",
+					Documentation: "The name of the integration to create.",
+				},
+			},
+		},
+		Flags: cmd.Flags{
+			Local: []*cmd.Flag{
+				{
+					Name:         "config-file",
+					DisplayValue: "CONFIG_FILE",
+					Description:  "File path to read integration config data.",
+					Value:        flagvalue.Simple("", &opts.ConfigFilePath),
+					Required:     true,
+				},
+			},
+		},
+		RunF: func(c *cmd.Command, args []string) error {
+			opts.IntegrationName = args[0]
+
+			if runF != nil {
+				return runF(opts)
+			}
+			return createRun(opts)
+		},
+	}
+
+	return cmd
+}
+
+type IntegrationConfig struct {
+	Version string
+	Type    IntegrationType
+	Details map[string]string
+}
+
+var (
+	TwilioKeys = []string{"account_sid", "api_key_secret", "api_key_sid"}
+	MongoKeys  = []string{"private_key", "public_key"}
+	AWSKeys    = []string{"audience", "role_arn"}
+	GCPKeys    = []string{"audience", "service_account_email"}
+)
+
+func createRun(opts *CreateOpts) error {
+	// Open the file
+	f, err := os.ReadFile(opts.ConfigFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to open config file: %w", err)
+	}
+
+	var i IntegrationConfig
+	err = yaml.Unmarshal(f, &i)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal config file: %w", err)
+	}
+
+	switch i.Type {
+	case Twilio:
+		missingField := validateDetails(i.Details, TwilioKeys)
+
+		if missingField != "" {
+			return fmt.Errorf("missing required field in the config file: %s", missingField)
+		}
+
+		body := &preview_models.SecretServiceCreateTwilioIntegrationBody{
+			Name:               opts.IntegrationName,
+			TwilioAccountSid:   i.Details["account_sid"],
+			TwilioAPIKeySecret: i.Details["api_key_secret"],
+			TwilioAPIKeySid:    i.Details["api_key_sid"],
+		}
+
+		resp, err := opts.PreviewClient.CreateTwilioIntegration(&preview_secret_service.CreateTwilioIntegrationParams{
+			Context:        opts.Ctx,
+			ProjectID:      opts.Profile.ProjectID,
+			OrganizationID: opts.Profile.OrganizationID,
+			Body:           body,
+		}, nil)
+
+		if err != nil {
+			return fmt.Errorf("failed to create Twilio integration: %w", err)
+		}
+
+		fmt.Fprintln(opts.IO.Err())
```

Comment:
thoughts on moving  these 2 fmt.Fprint lines out of the switch cases and before the return? since it can also be done with opts.IntegrationName, and should reduce lines a bit